hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -213,6 +221,29 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
     }
   }
 
+  /**
+   * Check whether expression of preCombine field have the same column name with target column.
+   *
+   * Merge expression cases that return true:
+   * 1) merge into .. on .. update set ts = s0.ts
+   * 2) merge into .. on .. update set ts = cast(s0.ts as int)
+   * 3) merge into .. on .. update set ts = s0.ts+1 (expressions like this whose sub node has the same column name with target)
+   * "ts" is preCombine field of h0.
+   */
+  private def isEqualToTargetPreCombineField(targetColumnName: String, sourceExpression: Expression): Boolean = {
+    val sourceColumnName = sourceDFOutput.map(_.name)
```

Comment:
This should be plural. Please update this in `isEqualToTarget` as well

Reply:
It'd be `sourceColumnNames`

Reply:
okay, got. Done.